src/net.cpp

```diff
@@ -589,14 +599,22 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes, bool& complete
     while (nBytes > 0) {
         // absorb network data
         int handled = m_deserializer->Read(pch, nBytes);
-        if (handled < 0) return false;
+        if (handled < 0){
+            LogPrint(BCLog::NET, "HEADER ERROR peer=%d\n", id);
+            return false;
+        }
 
         pch += handled;
         nBytes -= handled;
 
         if (m_deserializer->Complete()) {
             // decompose a transport agnostic CNetMessage from the deserializer
-            CNetMessage msg = m_deserializer->GetMessage(Params().MessageStart(), nTimeMicros);
+            Optional<CNetMessage> result = m_deserializer->GetMessage(nTimeMicros);
+            if(!result){
+                LogPrint(BCLog::NET, "HEADER ERROR peer=%d\n", id);
```

Comment:
I think a comment here to say "Message deserialization failed. Drop the message and log the failure, but don't disconnect the peer" would be useful

Reply:
Done